src/game/Board.java

```diff
@@ -0,0 +1,376 @@
+	 */
+	private void areaBlack(Intersec intersec) {
+		if (intersec.getMark() == Mark.U && !intersec.isChecked()) {
+			intersec.setBlackArea(true);
+			intersec.setChecked(true);
+			for (Intersec neighbour : intersec.getNeighbours()) {
+				areaBlack(neighbour);
+			}
+		}
+	}
+
+	/**
+	 * Gives a point to black for each black stone on the board. Calls the
+	 * areaBlack(Intersec intersec) method on each intersection with a black stone
+	 * on there.
+	 */
+	private void scoringWhite() {
+		for (int i = 0; i < intersecs.length; i++) {
+			if (intersecs[i].getMark() == Mark.W) {
+				whiteScore = whiteScore + 1;
+				for (Intersec neighbour : intersecs[i].getNeighbours()) {
+					areaWhite(neighbour);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Determines for all intersections in an empty area whether they belong to the
+	 * area of black (if so, their blackArea boolean is set to true). This is the
+	 * case when at least one black stone borders this empty area.
+	 */
+	private void areaWhite(Intersec intersec) {
+		if (intersec.getMark() == Mark.U && !intersec.isChecked()) {
+			intersec.setWhiteArea(true);
+			intersec.setChecked(true);
+			for (Intersec neighbour : intersec.getNeighbours()) {
+				areaWhite(neighbour);
+			}
+		}
+	}
+
+	/**
+	 * Determine the winner of the game. Return true if white wins, false if black
+	 * wins.
+	 */
+	public boolean determineWinner() {
+		countScore();
+		System.out.println(
+				"The score has been counted! White's score is " + whiteScore + ", black's score is " + blackScore);
+		return (whiteScore > blackScore);
+	}
+
+	/**
+	 * Checks whether the board is full (and hence the game is over).
+	 */
+	public boolean gameOver() {
+		int i = 0;
+		if (!firstStone) {
+			return false;
+		}
+		for (int row = 0; row < boardSize; row++) {
+			for (int col = 0; col < boardSize; col++) {
+				if (intersecs[i].getMark() != Mark.U) {
+					return false;
+				}
+				i++;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Stores all the board situations to check whether no previous board situation
+	 * is recreated (Ko-rule). This method should not be called when a player
+	 * passes!!!
+	 * 
+	 * @throws ExitProgram when the Ko-rule has been violated
+	 */
+	public void addBoardSituation(String boardSituation) throws ExitProgram {
+		if (boardSituations.contains(boardSituation)) {
+			throw new ExitProgram("The Ko rule has been violated!");
+		} else {
+			this.boardSituations.add(boardSituation);
+		}
+	}
+
+	/**
+	 * Getter method for the GUI. Useful for quitting a game.
+	 */
+	public GoGUIIntegrator getGUI() {
+		return g;
+	}
+
+	/**
+	 * Turns a Board into a String. Useful for debugging. Should be used to
+	 * communicate the contents of a board situation. Also used for determining
+	 * whether the Ko-rule is breached.
+	 */
+	@Override
+	public String toString() {
+		String s = "";
+		for (int i = 0; i < intersecs.length; i++) {
+			s = s + intersecs[i];
+		}
+		return s;
+	}
+
+}
```

Comment:
That's really a big class, maybe you could have extracted some functionality here.